dotnet/src/SemanticKernel.Abstractions/Filters/Prompt/PromptFilterContext.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Base class with data related to prompt rendering.
+/// </summary>
+[Experimental("SKEXP0005")]
+public abstract class PromptFilterContext
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PromptFilterContext"/> class.
+    /// </summary>
+    /// <param name="function">The <see cref="KernelFunction"/> with which this filter is associated.</param>
+    /// <param name="arguments">The arguments associated with the operation.</param>
+    /// <param name="metadata">A dictionary of metadata associated with the operation.</param>
+    internal PromptFilterContext(KernelFunction function, KernelArguments arguments, IDictionary<string, object?>? metadata)
```

Comment:
Shouldn't these be readonly dictionaries? Or is the intent that the handler can modify/augment the metadata?

Reply:
Good catch, updated!